revert: Temporarily revert some breaking changes (for 2.0.0-internal.6.x)

alexvy86 opened the pull request:
## Description

Additional reverts to ease transition to `2.0.0-internal.6.x`.

- Restore `IDisposable` and `EventForwarder` members on `PureDataObject` (partial revert of https://github.com/microsoft/FluidFramework/pull/16283).
- Restore `id` property on `IContainerContext` (partial revert of https://github.com/microsoft/FluidFramework/pull/16262)
- ~Restore `parent` property on `ISegment` (partial revert of https://github.com/microsoft/FluidFramework/pull/15985)~ trying to address this in a different way, so no reversal here in the end.

All of these reverts will be undone (i.e. changes re-applied) in `2.0.0-internal.7.0`.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

@ChumpChief , @anthony-murphy , would appreciate a check on your respective changes. I think the reverts for Matt's seem fine, but not so confident on Tony's, and reverting the whole PR felt like it might be overkill. But I couldn't quite track what was setting that `ISegment.parent`.

alexvy86 committed:
docs: Remove outdated docs from devtools-view README (#16835)

anthony-murphy committed:
Fix SharedString Event and Submit Order to Avoid Reentrancy (#16815)

This change primarily addresses a problem in sequence where the event
for a change is triggered before the op for the change is submitted.
This is a problem as if an application does reentrancy, specifically
creates more changes in the event for a change, it will lead to the
subsequent changes, made is the event, to be submitted before the
original change. This can result in document corruption.

We fix this issue by explicitly submitting before we fire the event.
Since we now submit on all events, it is also necessary to not submit in
some cases, specifically where there are stashed ops, as these may have
already been submitted.

Lastly, this change deprecates grouped ops, which also have reentrancy
concerns, specifically that if a change is made during the application
of a group op, that change cannot be correctly applied, as the group op
only has a single sequence number so there is no way to identify at
which sub-op the new change should be made after. Remote client will end
up processing the new change after the group op is complete, which is
not when the new change was generated, and can lead to eventual
consistency issues.

---------

Co-authored-by: Tony Murphy <anthonm@microsoft.com>
Co-authored-by: Wes Carlson <49205066+wes-carlson@users.noreply.github.com>

daesunp committed:
Add SharedTree fuzz tests to stress pipeline (#16773)

## Description

This PR adds the sharedTree fuzz tests to the stress pipeline.

agarwal-navin committed:
Rename reason to summarizeReason so that it is not overwritten on summarize failure (#16828)

The `reason` for summarization is overwritten by [this
code](https://github.com/microsoft/FluidFramework/blob/277f5b8e89b4c0c3dcdac95fe4b5c7063cfdb870/packages/runtime/container-runtime/src/summary/summaryGenerator.ts#L293)
if summarization fails. This means that the ``Summarizer_cancel` event
will not have the reason for summarization. It would be helpful to know
the reason for the summarization that fails.

This PR renames the summarize reason to `summarizeReason` instead of
just `reason` which is generic and can be overwritten.

alexvy86 committed:
Revert EventForwarder+IDisposable changes in PureDataObject (PR 16283)

Josmithr committed:
refactor(container-utils): Deprecate DeltaManagerProxyBase (#16813)

Deprecates DeltaManagerProxyBase. No replacement API is intended.

---------

Co-authored-by: Kajari Ghosh <ghoshkaj@gmail.com>

clarenceli-msft committed:
Couple the client build and test stability pipeline (#16707)

[AB#4808](https://dev.azure.com/fluidframework/internal/_workitems/edit/4808)

Since the client build pipeline yaml file is updated frequently, there
is a high risk of breaking the test stability pipeline due to the async
changes in Install/Build/Test steps. Moving the overlapped/shared parts
to separate template files will help to forcibly keep them in sync and
reduce future overhead.

The Test Stability pipeline with this update was run successfully on the
personal test branch:
https://dev.azure.com/fluidframework/internal/_build/results?buildId=180300&view=results

tylerbutler commented on .changeset/eighty-weeks-relate.md:
```diff
@@ -0,0 +1,13 @@
+PureDataObject temporarily extends EventForwarder and implements IDisposable again
+
+`PureDataObject` extends `EventForwarder` and implements `IDìsposable` again to ease the transition to `2.0.0-internal.6.x`.
+These changes will be undone in `2.0.0-internal.7.0.0`.
```
```suggestion
These interfaces will no longer be implemented on `PureDataObject` in version `2.0.0-internal.7.0.0`.
```

tylerbutler commented on packages/framework/aqueduct/src/data-objects/pureDataObject.ts:
```diff
@@ -175,4 +192,42 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
+	}
+
+	/**
+	 * @deprecated - 2.0.0-internal.5.2.0 - PureDataObject does not actually set up to forward events, and will not be an EventForwarder
```
```suggestion
	 * @deprecated 2.0.0-internal.5.2.0 - PureDataObject does not actually set up to forward events, and will not be an EventForwarder
```

no dashes after \@deprecated in TSDoc

alexvy86 committed:
Update .changeset/eighty-weeks-relate.md

Co-authored-by: Tyler Butler <tyler@tylerbutler.com>

alexvy86 committed:
Update packages/framework/aqueduct/src/data-objects/pureDataObject.ts

Co-authored-by: Tyler Butler <tyler@tylerbutler.com>

curtisman committed:
Remove auth to office feed (#16831)

Already switched to use a feed in our ADO instance, no more need to auth
to office feed.

alexvy86 committed:
fix: Move output directory for e2e-tests app build in devtool-browser-extension (#16847)

alexvy86 committed:
Revert removal of id in IContainerContext (PR 16262)

alexvy86 committed:
Revert removal of parent in ISegment (PR 15985)

alexvy86 committed:
Merge branch 'main' into more-reverts

tylerbutler commented on api-report/merge-tree.api.md:
```diff
@@ -405,6 +405,10 @@ export interface IMergeNodeCommon {
     // (undocumented)
     isLeaf(): this is ISegment;
     ordinal: string;
+    // Warning: (ae-forgotten-export) The symbol "IMergeBlock" needs to be exported by the entry point index.d.ts
+    //
+    // @internal @deprecated (undocumented)
+    parent?: IMergeBlock;
```
Probably need to export this.

alexvy86 committed:
Revert "Revert removal of parent in ISegment (PR 15985)"

This reverts commit 4a32719d1cc13c29fdbaf5d03825b3ebe1a11e8d.

alexvy86 committed:
Update changeset

Josmithr commented on packages/common/container-definitions/src/runtime.ts:
```diff
@@ -197,6 +197,17 @@ export interface IContainerContext {
+	 * WARNING: this id is meant for telemetry usages ONLY, not recommended for other consumption
+	 * This id is not supposed to be exposed anywhere else. It is dependant on usage or drivers
+	 * and scenarios which can change in the future.
+	 * @deprecated - 2.0.0-internal.5.2.0 - The docId is already logged by the IContainerContext.taggedLogger for
```
```suggestion
	 * @deprecated 2.0.0-internal.5.2.0 - The docId is already logged by the IContainerContext.taggedLogger for
```

`@deprecated` blocks don't want a `-` separator. See: https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines#deprecated

I would also recommend using an `{@link}` tag where we reference `IContainerContext.taggedLogger` for the generated docs.

Josmithr commented on packages/framework/aqueduct/src/data-objects/pureDataObject.ts:
```diff
@@ -53,6 +56,14 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
 	protected initializeP: Promise<void> | undefined;
 
+	/**
+	 * @deprecated - 2.0.0-internal.5.2.0 - PureDataObject does not provide a functioning built-in disposed flow.
```
```suggestion
	 * @deprecated 2.0.0-internal.5.2.0 - PureDataObject does not provide a functioning built-in disposed flow.
```

Josmithr reviewed (commented):
The changes seem reasonable to me, and the surrounding documentation seems clear enough, but I don't have a ton of context into what is / is not required to be reverted here.

I did leave a comment around the use of `@deprecated` tags though.

alexvy86 committed:
TSDoc improvements
